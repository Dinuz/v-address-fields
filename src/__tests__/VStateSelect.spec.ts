import Vue from 'vue'
// Utils
import { mount, MountOptions, Wrapper } from '@vue/test-utils'
// import { inspect } from 'util' // okay to delete this line if you don't use node inspect

// Component to be tested
import VAddressFields, { VStateSelect } from '..'

// Mocks
const vuetifyMocks = {
  $vuetify: {
    theme: {
      currentTheme: {
        error: '#ff0000',
      },
      dark: false,
    },
    lang: {
      t: (val: string) => val,
    },
  },
}

// Data necessary for tests
const foo = 'bar'
const messages = []

describe('VStateSelect', () => {
  describe('installer', () => {
    it('should register the v-state-select component', () => {
      Vue.use(VAddressFields)
      expect(Vue.options.components['v-state-select']).toBeTruthy()
    })
  })

  describe('component', () => {
    type Instance = InstanceType<typeof VStateSelect>
    let mountFunction: (options?: MountOptions<Instance>) => Wrapper<Instance>

    beforeEach(() => {
      mountFunction = (options?: MountOptions<Instance>) => {
        return mount(VStateSelect, {
          mocks: {
            ...vuetifyMocks,
          },
          ...options,
        })
      }
    })

    describe('initialization', () => {
      beforeEach(() => {
        // reset messages
        messages.length = 0
        // capture console messages to prevent them from cluttering the terminal
        const capture = (m: { toString: () => any }) => { messages.push(m.toString()) }
        jest.spyOn(global.console, 'error').mockImplementation(capture)
        jest.spyOn(global.console, 'warn').mockImplementation(capture)
        jest.spyOn(global.console, 'log').mockImplementation(capture)
      })

      it('should render component and match snapshot', () => {
        const wrapper = mountFunction()
        // replace the auto-generated `id` with one that matches
        const html = wrapper.html().replace(/div id="input-\d+"/, 'div id="input-1"')
        expect(html).toMatchSnapshot()
      })

      it('should have a property called <code>contiguousOnly</code> that defaults to <code>false</code>', () => {
        const wrapper = mountFunction()
        expect(wrapper.vm.contiguousOnly).toBeDefined()
        expect(wrapper.vm.contiguousOnly).toBe(false)
      })

      it('should have a property called <code>exclude</code> that defaults to an empty array', () => {
        const wrapper = mountFunction()
        expect(wrapper.vm.exclude).toBeDefined()
        expect(wrapper.vm.exclude).toStrictEqual([])
      })

      it('should have a property called <code>includeTerritories</code> that defaults to <code>false</code>', () => {
        const wrapper = mountFunction()
        expect(wrapper.vm.includeTerritories).toBeDefined()
        expect(wrapper.vm.includeTerritories).toBe(false)
      })
    })

    it('should generate a list of 51 "state" items', () => {
      const wrapper = mountFunction({ propsData: { foo } })
      expect(wrapper.vm.allItems).toBeDefined()
      expect(wrapper.vm.allItems).toHaveLength(51)
    })
  })
})
