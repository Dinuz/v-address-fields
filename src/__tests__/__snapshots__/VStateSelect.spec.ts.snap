// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VStateSelect component initialization should render component and match snapshot 1`] = `
<div class="v-input theme--light v-text-field v-select v-autocomplete v-state-select">
  <div class="v-input__control">
    <div role="combobox"
         aria-haspopup="listbox"
         aria-expanded="false"
         aria-owns="list-1"
         class="v-input__slot"
    >
      <div class="v-select__slot">
        <input id="input-1"
               type="text"
               autocomplete="off"
        >
        <div class="v-input__append-inner">
          <div class="v-input__icon v-input__icon--append">
            <i aria-hidden="true"
               class="v-icon notranslate material-icons theme--light"
            >
              $dropdown
            </i>
          </div>
        </div>
        <input type="hidden">
      </div>
      <div class="v-menu"
           offsety="true"
      >
      </div>
    </div>
    <div class="v-text-field__details">
      <div class="v-messages theme--light">
        <transition-group-stub tag="div"
                               name="message-transition"
                               class="v-messages__wrapper"
        >
        </transition-group-stub>
      </div>
    </div>
  </div>
</div>
`;
